packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -0,0 +1,248 @@
+    TAKE_WITH_CAMERA
+  }
+
+  public ImagePickerDelegate(Activity registrar, ImageResizer imageResizer) {
+    this.registrar = registrar;
+    this.imageResizer = imageResizer;
+    this.providerName = registrar.getPackageName() + ".flutter.image_provider";
+  }
+
+  public void chooseImageFromGallery(MethodCall methodCall, MethodChannel.Result result) {
+    setPendingMethodCallAndResult(methodCall, result);
+
+    if (!hasRequiredPermissions()) {
+      pendingPickImageOperation = ImagePickingOperationType.CHOOSE_FROM_GALLERY;
+      requestPermissions();
+      return;
+    }
+
+    launchPickImageFromGalleryIntent();
+  }
+
+  private boolean hasRequiredPermissions() {
+    boolean hasExternalStoragePermission =
+        ActivityCompat.checkSelfPermission(registrar, Manifest.permission.WRITE_EXTERNAL_STORAGE)
+            == PackageManager.PERMISSION_GRANTED;
+
+    boolean hasCameraPermission =
+        ActivityCompat.checkSelfPermission(registrar, Manifest.permission.CAMERA)
+            == PackageManager.PERMISSION_GRANTED;
+
+    return hasExternalStoragePermission && hasCameraPermission;
+  }
+
+  private void requestPermissions() {
+    ActivityCompat.requestPermissions(
+        registrar,
+        new String[] {Manifest.permission.CAMERA, Manifest.permission.WRITE_EXTERNAL_STORAGE},
+        REQUEST_IMAGE_PICKER_PERMISSIONS);
+  }
+
+  private void launchPickImageFromGalleryIntent() {
+    Intent pickImageIntent = new Intent(Intent.ACTION_GET_CONTENT);
+    pickImageIntent.setType("image/*");
+
+    registrar.startActivityForResult(pickImageIntent, REQUEST_CODE_CHOOSE_FROM_GALLERY);
+  }
+
+  public void takeImageWithCamera(MethodCall methodCall, MethodChannel.Result result) {
+    setPendingMethodCallAndResult(methodCall, result);
+
+    if (!hasRequiredPermissions()) {
+      pendingPickImageOperation = ImagePickingOperationType.TAKE_WITH_CAMERA;
+      requestPermissions();
+      return;
+    }
+
+    launchTakeImageWithCameraIntent();
+  }
+
+  private void launchTakeImageWithCameraIntent() {
+    Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+
+    if (intent.resolveActivity(registrar.getPackageManager()) != null) {
+      File imageFile = createTemporaryWritableImageFile();
+      pendingCameraImageUri = Uri.parse("file:" + imageFile.getAbsolutePath());
+
+      Uri imageUri = FileProvider.getUriForFile(registrar, providerName, imageFile);
+      intent.putExtra(MediaStore.EXTRA_OUTPUT, imageUri);
+      grantUriPermissions(intent, imageUri);
+
+      registrar.startActivityForResult(intent, REQUEST_CODE_TAKE_WITH_CAMERA);
+    }
+  }
+
+  private File createTemporaryWritableImageFile() {
+    String filename = UUID.randomUUID().toString();
+    File storageDirectory = registrar.getExternalFilesDir(Environment.DIRECTORY_PICTURES);
+    File image;
+
+    try {
+      image = File.createTempFile(filename, ".png", storageDirectory);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
```

Comment:
Why not bubble up the more specific IOException?

Reply:
See [this comment](https://github.com/flutter/plugins/pull/453#discussion_r178806771).